sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -187,6 +189,14 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
         "read files of Hive data source directly.")
     }
 
+    // SPARK-23817 Since datasource V2 didn't support reading multiple files yet,
+    // ORC V2 is only used when loading single file path.
+    val allPaths = CaseInsensitiveMap(extraOptions.toMap).get("path") ++ paths
+    val orcV2 = OrcDataSourceV2.satisfy(sparkSession, source, allPaths.toSeq)
+    if (orcV2.isDefined) {
+      option("path", allPaths.head)
+      source = orcV2.get
+    }
```

Comment:
It seems weird that DataFrameReader is modified here. Will DataSourceV2 implementations generally need to modify DataFrameReader, or is it just a temporary hack because of the mentioned lack of support? In the latter case, is there a plan to add this support soon?

Reply:
This is temporary hack. I think @cloud-fan will create a PR to support reading multiple files recently.

Reply:
yea it's a temporary hack. We will support multi-path soon.

Reply:
What about bucketed reads? WIll they need a similar change here, or is that lack of support handled elsewhere? (Or am I misunderstanding something about that part of the description - I'm not super familiar with the ORC source)

Reply:
How does v2 not support reading multiple files?

Reply:
because we need to define how to pass multiple paths via options. I have a PR to fix it, I'll bring it update to date.